src/Maintenance/Maintenance.Test/ScenarioTests/MaintenanceTests.ps1

```diff
@@ -12,6 +12,93 @@
+        Assert-AreEqual $maintenanceConfiguration.Name $maintenanceConfigurationName
+        Assert-AreEqual $maintenanceConfiguration.Location $location
+        Assert-AreEqual $maintenanceConfiguration.MaintenanceScope $maintenanceScope
+        Assert-AreEqual $maintenanceConfiguration.Visibility $visibility
+        Assert-AreEqual $maintenanceConfiguration.StartDateTime $startDateTime
+        Assert-AreEqual $maintenanceConfiguration.Timezone $timezone
+        Assert-AreEqual $maintenanceConfiguration.RecurEvery $recurEvery
+        Assert-AreEqual $maintenanceConfiguration.Duration $duration
+        Assert-AreEqual $maintenanceConfiguration.ExpirationDateTime $expirationDateTime
+
+        # Retrieve and Validate Maintenance Configuration with ResourceGroupName
+        $retrievedMaintenanceConfiguration = Get-AzMaintenanceConfiguration -ResourceGroupName $resourceGroupName -Name $maintenanceConfigurationName
+        
+        # Validate the type of the returned object
+        if ($retrievedMaintenanceConfiguration -isnot [System.Collections.IList]) {
+            Write-Host "The retrieved maintenance configuration with ResourceGroupName is not of type list."
+        } else {
+            throw "The retrieved maintenance configuration with ResourceGroupName is of type list."
+        }
+
+        Assert-AreEqual $retrievedMaintenanceConfiguration.Name $maintenanceConfigurationName
+        Assert-AreEqual $retrievedMaintenanceConfiguration.Location $location
+        Assert-AreEqual $retrievedMaintenanceConfiguration.MaintenanceScope $maintenanceScope
+        Assert-AreEqual $retrievedMaintenanceConfiguration.Visibility $visibility
+        Assert-AreEqual $retrievedMaintenanceConfiguration.StartDateTime $startDateTime
+        Assert-AreEqual $retrievedMaintenanceConfiguration.Timezone $timezone
+        Assert-AreEqual $retrievedMaintenanceConfiguration.RecurEvery $recurEvery
+        Assert-AreEqual $retrievedMaintenanceConfiguration.Duration $duration
+        Assert-AreEqual $retrievedMaintenanceConfiguration.ExpirationDateTime $expirationDateTime
+
+        # Retrieve Maintenance Configuration without ResourceGroupName and validate it returns a list
+        $maintenanceConfigurations = Get-AzMaintenanceConfiguration -Name $maintenanceConfigurationName
+
+        # Validate that the returned object is a list
+        if ($maintenanceConfigurations -isnot [System.Collections.IList]) {
+            Write-Host "The retrieved maintenance configuration without ResourceGroupName is of type list."
+        } else {
+            throw "The retrieved maintenance configuration without ResourceGroupName is not of type list."
+        }
+
+        # As a final check, also verify that Update-Configuration can accept the inpu
+        try{
+            Update-AzMaintenanceConfiguration -ResourceGroupName $resourceGroupName -Name $maintenanceConfigurationName -Configuration $retrievedMaintenanceConfiguration
```

Comment:
Could you try to pass values via pipeline in test cases? like $retrievedMaintenanceConfiguration | Update-AzMaintenanceConfiguration -ResourceGroupName $resourceGroupName -Name $maintenanceConfigurationName, that's the case of reported issue.

Reply:
Updated this in the next commit. Thanks!